pkg/client-sdk/covenantless_client.go

```diff
@@ -277,6 +277,11 @@ func (a *covenantlessArkClient) InitWithWallet(ctx context.Context, args InitWit
 }
 
 func (a *covenantlessArkClient) listenForArkTxs(ctx context.Context) {
+	if err := a.refreshTxDb(ctx); err != nil {
+		log.WithError(err).Error("failed to refreshTxDb")
+		return
+	}
```

Comment:
`listenForArkTxs` runs as a go routine, and this way we would log an error and stop its execution, but the SDK client would still be returned as usable to the user. Better to move this outside, this way if this process fails we return error to the user.

Reply:
Done